nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/connect/LauncherRegisterPromptHelper.java

```diff
@@ -0,0 +1,183 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nuxeo
+ */
+
+package org.nuxeo.launcher.connect;
+
+import java.io.Console;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.artofsolving.jodconverter.util.PlatformUtils;
+import org.nuxeo.connect.connector.NuxeoClientInstanceType;
+import org.nuxeo.connect.data.ConnectProject;
+import org.nuxeo.launcher.config.ConfigurationException;
+
+import com.sun.istack.internal.NotNull;
+
+/**
+ * Helper methods to prompt different things to the user if the shell is in interactive mode.
+ *
+ * @author <a href="mailto:ak@nuxeo.com">Arnaud Kervern</a>
+ * @since 8.3
+ */
+public class LauncherRegisterPromptHelper {
+    private static final Log log = LogFactory.getLog(LauncherRegisterPromptHelper.class);
+
+    private static final int PAGE_SIZE = 20;
+
+    public static final String CONNECT_TC_URL = "https://www.nuxeo.com/en/services/nuxeo-trial-tc";
+
+    public static String promptMail() {
+        return prompt("Mail Adress: ", "^.+@.+\\..{2,4}", "Mail is invalid.");
+    }
+
+    public static String promptUsername() {
+        return prompt("Username: ", "^\\w+", "Username cannot be empty.");
+    }
+
+    public static String promptCompany() {
+        return prompt("Company: ", "^\\w+", "Company cannot be empty.");
+    }
+
+    public static String promptProjectName() {
+        return prompt("Project name: ", "^(?:[-\\w]+|)$",
+                "Project name can only contain alphanumeric chars and dashes.");
+    }
+
+    public static String promptDescription() {
+        return prompt("Description: ", "", null);
+    }
+
+    protected static String prompt(String message, String regex, String error) {
+        if (StringUtils.isBlank(regex)) {
+            regex = ".*";
+        }
+
+        Console console = System.console();
+        String value = null;
+        do {
+            if (value != null) {
+                console.printf(error + "\n", value);
+            }
+            value = console.readLine(message);
+        } while (!value.matches(regex));
+
+        return value;
+    }
+
+    public static String promptPassword(boolean confirmation) throws IOException, ConfigurationException {
+        String pwd = promptPassword("Please enter your password: ");
+        if (confirmation) {
+            String pwdVerification = promptPassword("Please re-enter your password: ");
+            if (!pwd.equals(pwdVerification)) {
+                throw new ConfigurationException("Passwords do not match.");
+            }
+        }
+        return pwd;
+    }
+
+    public static NuxeoClientInstanceType promptInstanceType() {
+        NuxeoClientInstanceType ret;
+        Console console = System.console();
+
+        do {
+            String s = console.readLine("Instance type (dev|preprod|prod) (default: dev): ");
```

Comment:
Format usage on nuxeoctl for the default value on prompt is: 
`Do you want to continue (yes/no)? [yes]`